### PR TITLE
IRC nick blacklist feature

### PR DESCRIPTION
**Pull request by peterix**

### CODE REVIEW

File: lib/bot.js [hunk 1]
```diff
@@ -33,6 +33,13 @@ class Bot {
     this.ircNickColor = options.ircNickColor !== false; // default to true
     this.channels = _.values(options.channelMapping);
 
+    /* Load blacklist - list of regexps of irc nicks that should be ignored by the relay */
+    this.ircBlacklist = [];
+    var ircBlacklistStrings = options.ircNickBlacklist || [];
+    ircBlacklistStrings.forEach(ircBlacklistString => {
```
Comment (RX14) [inline lib/bot.js:39]:
``` js
this.ircBlacklist = ircBlacklistString.map(string => new RegExp(string, ""))
```

would have been shorter

File: lib/bot.js [hunk 2]
```diff
@@ -171,6 +178,14 @@ class Bot {
   sendToDiscord(author, channel, text) {
     const discordChannelName = this.invertedMapping[channel.toLowerCase()];
     if (discordChannelName) {
+      for (let index = 0; index < this.ircBlacklist.length; ++index) {
+        let blackListItem = this.ircBlacklist[index];
+        if(blackListItem.test(author)) {
+          logger.debug('Ignoring message from IRC user', author, 'in channel', channel);
+          return;
```
Comment (RX14) [inline lib/bot.js:185]: Sucks that you can't return from lambdas

**Review by RX14 — APPROVED** [review]
I'll merge it anyway cause JS is trash anyway

File: lib/bot.js (hunk 2 above)
Comment (peterix) [reply lib/bot.js:185]: I know. I used the forEach thing and it didn't work...

**Pull request merged**